datatypes/utils.cpp

```diff
@@ -71,10 +71,10 @@ quint64 Utils::getTimeStamp()
     return data;
 }
 
-quint64 Utils::getTimeStamp(const struct timeval *tp)
+quint64 Utils::getTimeStamp(const struct input_event *ie)
 {
-    quint64 data = tp->tv_sec;
+    quint64 data = ie->input_event_sec;
```

Comment:
I wonder if this bit should be prepared for "input_event_sec" not having been defined?

AFAICT this 'input_event_sec'  -> linux/input.h -> kernel-headers package -> subject to kernel version, and some devices have old kernels...

A quick smoke-test in my xperia 10 target yields:

    error: ‘const struct input_event’ has no member named ‘input_event_sec’

Reply:
Ugh, old kernels. If only everything was mainlined properly...

I suppose we should define it if it´s not available. Something like

```
#ifndev input_event_sec
#define input_event_sec time.tv_sec
#endif
#ifndev input_event_usec
#define input_event_usec time.tv_usec
#endif
```

Not sure how to properly do this.

But yeah it´s definitely dependent on kernel versions, I´m not sure which version introduced it though.

Reply:
Apparently my assumption that kernel-headers version needs to be in sync with kernel version was incorrect. So this compilation problem might go away if/when kernel-headers version used in sfos gets updated.

Reply:
Oh interesting. Any ETA on that?
I can keep this patch downstream in postmarketOS for now till SFOS updates.

Reply:
I've been told that updating kernel-headers is probably not going to happen in immediate future... (sizable testing effort etc)

After a quick look at input.h history: before input_event_sec got defined, there was always just "struct timeval time" in the input event structure.

So, maybe one could utilize workaround like: if input_event_sec & are defined, use them. Else use time.tv_sec & co (as it was doing before these changes). Something like:

```
quint64 Utils::getTimeStamp(const struct input_event *ie)
{
#ifdef input_event_sec
    quint64 data = ie->input_event_sec;
    data = data * 1000000;
    data = ie->input_event_usec + data;
#else
    /* Fallback used in case of old kernel-headers package that
     * does not define input_event_sec etc in linux/input.h
     */
    quint64 data = ie->time.tv_sec;
    data = data * 1000000;
    data = ie->time.tv_usec + data;
#endif
    return data;
}

```